File: BookExamples/54_SendBroadcast/app/src/main/java/com/ebookfrenzy/sendbroadcast/MyReceiver.java
package com.ebookfrenzy.sendbroadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

public class MyReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String intentData="";
        Bundle extras = intent.getExtras();

        //THIS PREVENTS THE STRING TO BE ADDED IF POWER DISCONNECT IS SENT
        if(extras.getString("MyData") != null){
            intentData = extras.getString("MyData");
        }
        else{
            intentData = "";
        }

        String message = "Broadcast intent detected " + intent.getAction() + intentData;

        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
    }
}

File: class_book_examples/chapter62_SendBroadcast/app/src/main/java/com/ebookfrenzy/sendbroadcast/MyReceiver.kt
package com.ebookfrenzy.sendbroadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast

class MyReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {

        val extras = intent.extras?:return


        var name = ""

        extras.getString("name").let {
            if (it != null) {
                name = it
            }
        }


        var message = "Broadcast intent detected " + intent.action + "  " + name

        //var message = "Broadcast intent detected " + intent.action

        if(intent.action == "android.intent.action.ACTION_POWER_DISCONNECTED"){
            message = "it works"
        }

        Toast.makeText(context, message,Toast.LENGTH_LONG).show()
    }
}